ffi/src/sspi/sec_winnt_auth_identity.rs

```diff
@@ -582,6 +583,38 @@ pub unsafe fn unpack_sec_winnt_auth_identity_ex2_w_sized(
         ));
     }
 
+    // try to collect credentials for the emulated smart card
+    #[cfg(feature = "scard")]
+    if username.contains(&b'@') {
+        use winscard::SmartCardInfo;
+
+        use crate::utils::str_encode_utf16;
+        use crate::winscard::scard_context::{DEFAULT_CARD_NAME, MICROSOFT_DEFAULT_CSP};
+
+        match SmartCardInfo::try_from_env() {
+            Ok(smart_card_info) => {
+                // remove null
+                let new_len = password.as_ref().len() - 2;
+                password.as_mut().truncate(new_len);
```

Comment:
nitpick: I recommend writing ["proper sentences"](https://github.com/Devolutions/IronRDP/blob/60080191aa34a47e075ee7692238f34220c6092f/STYLE.md#inline-code-comments-are-proper-sentences) to  create a more appropriate frame of mind, and the expectation is to write down more of the context you had in mind when writing this code.

For instance, in this case what I really want to know as someone non familiar with the smart card protocol is "why we need to remove the null terminator".

This comment can be applied at several places, but I leave you decide if and how much you do it.

(All that being said, the code you wrote is generally well documented, thank you again.)

Reply:
I agree with you. Can you check the last commit? I updated some comments

Reply:
It’s looking much better! For this place specifically, I would like something more like that:

> The "pin" buffer of the SmartCardIdentityBuffers structure must not be NULL-terminated because / as specified in […] (protocol spec reference? empirical observation?). 
> Since the password data returned by CredUnPackAuthenticationBufferW is a NULL-terminated wide C string, we need to remove the last two bytes.

The conversion itself is a detail, and what is important is the surroundings, the things we can’t see just by looking at the code.

(Again, I’m really being pedantic here.)

Reply:
> Again, I’m really being pedantic here.

I can understand you and I like it 😃.

I added more comprehensive explanations